src/server/controllers/home-block.controller.ts

```diff
@@ -36,17 +43,35 @@ export const getHomeBlocksHandler = async ({
 }: {
   ctx: Context;
   input: GetHomeBlocksInputSchema;
-}) => {
+}): Promise<HomeBlockWithData[]> => {
+  const { ownedOnly } = input || {};
```

Comment:
Since input is required, I believe we don't need the fallback here
```suggestion
  const { ownedOnly } = input;
```